### PR TITLE
test(io_tune): verify io_tune is only called once

### PR DESCRIPTION
**Pull request by fruch**
during the life-cycle of a pod, the results of the io_tune
should be cached, and not run on every restart of the pod.

Ref: https://github.com/scylladb/scylla-operator/issues/146
Trello: https://trello.com/c/kFSyujoc

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster_k8s/mini_k8s.py [hunk 1]
```diff
@@ -216,7 +216,10 @@ def local_kubectl_version(self):  # pylint: disable=no-self-use
 
     def docker_pull(self, image):
         LOGGER.info("Pull `%s' to docker environment", image)
-        self.remoter.run(f"docker pull -q {image}")
+        if image.endswith('latest') or not self.remoter.run(f"docker images -q {image}").stdout.strip():
```
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/cluster_k8s/mini_k8s.py:219]: Not directly, but it's a small fix I did while trying to run those test locally (it's on it's own commit)

[the remark this answers is not in the document]
Comment (vponomaryov) [reply sdcm/cluster_k8s/mini_k8s.py:219]: This change doesn't fix anything. `docker pull` command doesn't pull images if they exist.

The problem is related to the images which are not tracked by this command and get pulled directly to the `KinD` each time we create new cluster. 
KinD doesn't reuse SCT runner's docker image registry. It uses it's own where we should explicitly load images from SCT runner.

I will create PR for caching other images.

**Review by vponomaryov — CHANGES_REQUESTED** [review]

File: sdcm/cluster_k8s/mini_k8s.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/cluster_k8s/mini_k8s.py:219]: `docker pull` communicate with the docker registry to check if the image was updated. checking local is quicker.
we are doing the same in hydra.

[the remark this answers is not in the document]
Comment (vponomaryov) [reply sdcm/cluster_k8s/mini_k8s.py:219]: The commit message is completely different than `avoid API call and make local service call`.
It states that it fixes duplicate docker pulls, but it is not.
If simple API call times out then there is wider problem like broken internet connection of down docker services.

[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/cluster_k8s/mini_k8s.py:219]:
```
❯ time( docker pull calico/node:v3.23.0 ) 
v3.23.0: Pulling from calico/node
Digest: sha256:4763820ecb4d8e82483a2ffabfec7fcded9603318692df210a778d223a4d7474
Status: Image is up to date for calico/node:v3.23.0
docker.io/calico/node:v3.23.0

real    0m2.166s
user    0m0.026s
sys     0m0.033s

❯ time( docker images calico/node:v3.23.0 ) 
REPOSITORY    TAG       IMAGE ID       CREATED       SIZE
calico/node   v3.23.0   593794e30dc6   2 weeks ago   201MB

real    0m0.062s
user    0m0.043s
sys     0m0.027s
```

for hydra it was a big difference, here I guess there are bigger `fishes` to fry first, to get it working a bit quicker.

[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/cluster_k8s/mini_k8s.py]: removed this commit

File: sdcm/sct_config.py [hunk 2]
```diff
@@ -1104,7 +1104,7 @@ class SCTConfiguration(dict):
              help="options will be used for enable encryption at-rest for tables"),
 
         dict(name="logs_transport", env="SCT_LOGS_TRANSPORT", type=str,
-             help="How to transport logs: rsyslog, ssh or docker", choices=("rsyslog", "ssh", "docker", "syslog-ng")),
+             help="How to transport logs: rsyslog, ssh or docker", choices=("rsyslog", "ssh", "docker", "syslog-ng", "kubectl")),
```
Comment (vponomaryov) [inline sdcm/sct_config.py:1107]: I don't think we should update it. `kubectl` is used in parallel to one of the main log transports (syslogng or rsyslog).

Comment (fruch) [reply sdcm/sct_config.py:1107]: removed

**Pull request merged**